src/main/splitting/tasks_tests_handler.py

```diff
@@ -0,0 +1,290 @@
+        call_args = ['g++', '-o', compiled_file_path, compiled_file + '.' + extension]
+    elif language == LANGUAGE.KOTLIN.value:
+        compiled_file_path = compiled_file + '.jar'
+        call_args = ['kotlinc', compiled_file + '.' + extension, '-include-runtime', '-d',
+                     compiled_file_path]
+    else:
+        raise ValueError('Language is not defined')
+
+    return call_args
+
+
+def __create_source_code_file(source_code: str, language=LANGUAGE.PYTHON.value,
+                              source_file_name=SOURCE_OBJECT_NAME):
+    if language == LANGUAGE.JAVA.value:
+        source_file_name = __get_java_class(source_code)
+    create_file(source_code, get_extension_by_language(language), __get_compiled_file(source_file_name))
+    return source_file_name
+
+
+def create_in_and_out_dict(tasks: list):
+    in_and_out_files_dict = {}
+    for task in tasks:
+        files = next(os.walk(TASKS_TESTS_PATH + task))[2]
+        in_and_out_files_dict[task] = __get_in_and_out_files(files)
+
+    return in_and_out_files_dict
+
+
+def is_python_file_correct(filename: str):
+    source = open(filename, 'r').read() + '\n'
+    if check_python_file_by_mypy(filename):
+        try:
+            code = compile(source, filename, 'exec')
+            signal.alarm(consts.MAX_SECONDS_TO_WAIT_TEST)
+            eval(code, {})
+            return True
+        except TimeoutException:
+            # It means that eval didn't throw any exceptions, but time is out for example because of input waiting
+            log.info("Time is out")
+            return True
+        except:
+            return False
+    return False
+
+
+def check_python_file_by_mypy(file_name: str):
+    call_args = ['mypy', file_name]
+    is_valid = __compile_program(call_args)
+    return is_valid
+
+
+def is_source_file_correct(source_file: str, language=LANGUAGE.PYTHON.value):
+    if language == LANGUAGE.PYTHON.value:
+        is_correct = is_python_file_correct(__get_compiled_file(source_file) + "." + get_extension_by_language(language))
+    else:
+        compiling_args = __get_args_for_compiling_program(language, source_file)
+        is_correct = __compile_program(compiling_args)
+    log.info("Source code is correct: " + str(is_correct))
+    return is_correct
+
+
+def check_before_tests(source_file: str, source_code: str, tasks: list, language=LANGUAGE.PYTHON.value):
+    test_results = []
+    need_to_run_tests = True
+    rate = consts.TEST_RESULT.CORRECT_CODE.value
+
+    # not to check incorrect fragments
+    if not is_source_file_correct(source_file, language):
+        need_to_run_tests = False
+        rate = consts.TEST_RESULT.INCORRECT_CODE.value
+        test_results = [rate] * len(tasks)
+
+    # not to check too small fragments because they cannot return true anyway
+    elif len(source_code) < consts.LANGUAGE_TO_MIN_SYMBOLS[language]:
+        log.info("Code fragment is too small")
+        need_to_run_tests = False
```

Comment:
I think we can define a function for the code fragment:
`need_to_run_tests = False`
`test_results = [rate] * len(tasks)`
Because we use it 3 times and it is "default values" for these cases